tests/test_gearbox.py

```diff
@@ -0,0 +1,143 @@
+# This test suite covers the functionality of gearbox feature
+import time
+import os
+import pytest
+from swsscommon import swsscommon
+from dvslib.dvs_database import DVSDatabase
+from dvslib.dvs_common import PollingConfig, wait_for_result
+
+# module specific dvs env variables
+DVS_ENV = ["HWSKU=brcm_gearbox_vs"]
+
+class Gearbox(object):
+    def __init__(self, dvs):
+        db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+        t = swsscommon.Table(db, "_GEARBOX_TABLE")
+        assert t
+        sr = t.getTableNameSeparator()
+
+        # "_GEARBOX_TABLE:phy:1"
+        # "_GEARBOX_TABLE:phy:1:ports:0"
+        # "_GEARBOX_TABLE:phy:1:lanes:200"
+        self.phys = {}
+        phy_table  = swsscommon.Table(db, sr.join([t.getKeyName(""), "phy"]))
+        for i in [x for x in phy_table.getKeys() if sr not in x]:
+            (status, fvs) = phy_table.get(i)
+            assert status == True
+            self.phys[i] = {"attrs" : dict(fvs)}
+
+            port_table = swsscommon.Table(db, sr.join([phy_table.getKeyName(i), "ports"]))
+            port_list = [x for x in port_table.getKeys() if sr not in x]
+            self.phys[i]["port_table"] = port_table
+            self.phys[i]["ports"] = {}
+            for j in port_list:
+                (status, fvs) = port_table.get(j)
+                assert status == True
+                self.phys[i]["ports"][j] = dict(fvs)
+
+            lane_table = swsscommon.Table(db, sr.join([phy_table.getKeyName(i), "lanes"]))
+            lane_list = [x for x in lane_table.getKeys() if sr not in x]
+            self.phys[i]["lanes"] = {}
+            for j in lane_list:
+                (status, fvs) = lane_table.get(j)
+                assert status == True
```

Comment:
do we need to wait here?

Reply:
Do not need. In dvs creation, it has verified all services ready. The table GEARBOX_TABLE in APP_DB has been created at one time by gearsyncd. In case some error happens, the assertion statements are enough.